src/block_cache/block_chunkcache.c

```diff
@@ -517,6 +521,31 @@ __wt_chunkcache_setup(WT_SESSION_IMPL *session, const char *cfg[], bool reconfig
 #endif
     }
 
+    WT_RET(__wt_config_gets(session, cfg, "chunk_cache.pinned", &cval));
+    __wt_config_subinit(session, &targetconf, &cval);
+    for (cnt = 0; (ret = __wt_config_next(&targetconf, &k, &v)) == 0; ++cnt)
+        ;
+    WT_RET_NOTFOUND_OK(ret);
+
+    if (cnt != 0) {
+        WT_RET(__wt_calloc_def(session, cnt + 1, &pinned_objects));
+        __wt_config_subinit(session, &targetconf, &cval);
+        for (cnt = 0; (ret = __wt_config_next(&targetconf, &k, &v)) == 0; ++cnt) {
+            if (!WT_PREFIX_MATCH(k.str, "table:"))
+                WT_RET_MSG(session, EINVAL,
```

Comment:
I think these early returns can leak `pinned_objects`. Rather than cleaning it up, what if we make `pinned_objects` visible earlier - so, straight after the `__wt_calloc(..., pinned_objects)` we say `chunkcache->pinned_objects = pinned_objects`? That way the memory is always reachable, even if we don't necessarily free it in this function.

Reply:
I was planning to do this earlier. However, if this function returns failure, we would fail to open WiredTiger and exit, so it shouldn't matter. Or Is this a case of good coding practice?

Reply:
Yep - memory leaks are bad even on failure paths, we shouldn't rely on the higher-level code falling over. (For example, we'll support reconfigure before too long, and then this would be a real issue.)

Reply:
Updated in [c2ff55f](https://github.com/wiredtiger/wiredtiger/pull/9415/commits/c2ff55faa48439b3c0664fea3147de7a881994f8)